x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engine_analytics.tsx

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useEffect } from 'react';
+
+import { useActions, useValues } from 'kea';
+
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiPanel,
+  EuiSpacer,
+  EuiSuperDatePicker,
+  EuiText,
+  useEuiTheme,
+} from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+import { ANALYTICS_PLUGIN } from '../../../../../common/constants';
+import { KibanaLogic } from '../../../shared/kibana';
+
+import { EuiButtonEmptyTo } from '../../../shared/react_router_helpers';
+
+import { EngineAnalyticsCard } from './engine_analytics_card';
+import { EngineAnalyticsLogic } from './engine_analytics_logic';
+import { EngineAnalyticsLens } from './engines_lens/engine_analytics_lens';
+import { EngineAnalyticsLensLogic } from './engines_lens/engine_analytics_lens_logic';
+
+export const EngineAnalytics: React.FC = () => {
+  const {
+    euiTheme: { colors: colors },
+  } = useEuiTheme();
+
+  const {
+    isNoResultsCardVisible,
+    queriesCount,
+    queriesCountPercentage,
+    noResults,
+    noResultsPercentage,
+    startDate,
+    endDate,
+    time,
+  } = useValues(EngineAnalyticsLogic);
+  const {
+    setNoResultsCardVisible,
+    setQueriesCardVisible,
+    setStartDate,
+    setEndDate,
+    setSearchSessionId,
+    setTimeChange,
+  } = useActions(EngineAnalyticsLogic);
+
+  const { xyAttributes, metricAttributesQueries, metricAttributesNoResults } =
+    useValues(EngineAnalyticsLensLogic);
+  const { getLensAreaAttributes, getLensMetricAttributes } = useActions(EngineAnalyticsLensLogic);
+
+  const { data, defaultDataView, formula } = useValues(KibanaLogic);
+
+  useEffect(() => {
+    setSearchSessionId(data?.search?.session.start());
+    getLensMetricAttributes({
```

Comment:
question: If I'm following this logic correctly (and I may well not be), we're using these effects to set data in Kea, and then retrieving that same data again? I comment on this elsewhere as well, but I'm wondering whether we can't just keep all (or at least most) of that logic local to this file instead of pushing it out to Kea, because we're really not sharing most (or any) of this logic across components.

Reply:
Done. Moved all the lens logic to this file -  [engine_analytics.tsx](https://github.com/elastic/kibana/blob/8364a5088dbfcdf2133dddfad493c306571e4680/x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engine_analytics.tsx#L64) component